dex/config/config.go

```diff
@@ -30,19 +20,22 @@ func options(cfgFile *ini.File) map[string]string {
 	return options
 }
 
-// Unmapify parses config options from the provided settings map into the
-// specified interface.
-func Unmapify(settings map[string]string, obj interface{}) error {
-	cfgData := optionsMapToINIData(settings)
-	return Parse(cfgData, obj)
+// Parse returns a collection of all key-value options in the provided config
+// file path or []byte data.
+func Parse(cfgPathOrData interface{}) (map[string]string, error) {
+	cfgFile, err := ini.Load(cfgPathOrData)
+	if err != nil {
+		return nil, err
+	}
+	return options(cfgFile), nil
 }
 
-// Parse parses config options from the provided config file path or []byte
+// ParseInto parses config options from the provided config file path or []byte
 // data into the specified interface.
-// If the config has section headers, the config options are first read into
-// a map, then converted to []byte before being parsed. Otherwise `obj` would
-// not be modified with any data from the config.
-func Parse(cfgPathOrData, obj interface{}) error {
+// If the config has section headers, the config options are first read into a
+// map, then converted to []byte before being parsed. Otherwise `obj` would not
+// be modified with any data from the config.
```

Comment:
How about this instead?

```
for _, section := range cfgFile.Sections() {
	err := section.MapTo(obj)
	if err != nil {
		return err
	}
}
return nil
```

Reply:
Looks cleaner, will have to test to be sure it produces the desired effect of parsing options from all groups into the obj.